misc-scripts/xref_mapping/XrefParser/RefSeqCoordinateParser.pm

```diff
@@ -23,405 +23,302 @@ use strict;
+    XM => 'RefSeq_mRNA_predicted',
+    XR => 'RefSeq_ncRNA_predicted',
+    NP => 'RefSeq_peptide',
+    XP => 'RefSeq_peptide_predicted',
+  };
+
+  my $source_ids; 
+  while (my ($source_prefix, $source_name) = each %{$source_names}) {
+    $source_ids->{$source_prefix} = $self->get_source_id_for_source_name( $source_name, 'otherfeatures', $dbi )
   }
-  if($file =~ /ofdbname[=][>](\S+?)[,]/){
-    $ofdbname = $1;
-  }
-  if($file =~ /ofpass[=][>](\S+?)[,]/){
-    $ofpass = $1;
+
+  if ($verbose) {
+    while (my ($source_prefix, $source_name) = each %{$source_names}) {
+      print "$source_name source ID = $source_ids->{$source_prefix}\n";
+    }
   }
- 
-  my $registry = "Bio::EnsEMBL::Registry";
 
-  #get the species name
+  # get the species name
   my %id2name = $self->species_id2name($dbi);
-  if (defined $species_name) { push @{$id2name{$species_id}}, $species_name; }
-  if (!defined $id2name{$species_id}) { next; }
-  $species_name = $id2name{$species_id}[0];
-
-  my $core_dba;
-  my $otherf_dba;
-
-  if (defined $project && $project eq 'ensembl') {
-# Can use user-defined database
-      if (defined $host) {
-          $core_dba = Bio::EnsEMBL::DBSQL::DBAdaptor->new(
-              '-host'     => $host,
-              '-user'     => $user,
-              '-pass'     => $pass,
-              '-dbname'   => $dbname,
-              '-port'     => $port,
-              '-species'  => $species_name.$host,
-              '-group'    => 'core',
-       );
-      } else {
-# Else, database should be on staging
-      $registry->load_registry_from_multiple_dbs(
-          {
-              -host    => 'mysql-ens-sta-1',
-	      '-port'    => 4519,
-              -user    => 'ensro',
-          },
-       );
-      $core_dba = $registry->get_DBAdaptor($species_name,'core');
-      }
-      if (defined $ofhost) {
-# Can use user-defined database
-          $otherf_dba = Bio::EnsEMBL::DBSQL::DBAdaptor->new(
-              '-host'     => $ofhost,
-              '-user'     => $ofuser,
-              '-pass'     => $ofpass,
-              '-port'     => $ofport,
-              '-dbname'   => $ofdbname,
-              '-species'  => $species_name,
-              '-group'    => 'otherfeatures',
-       );
-       $otherf_dba->dnadb($core_dba);
-      } else {
-# Else database should be on staging
-      $registry->load_registry_from_multiple_dbs( 
-	  {
-	      -host    => 'mysql-ens-sta-1',
-	      '-port'    => 4519,
-	      -user    => 'ensro',
-	  },
-       );
-      $otherf_dba = $registry->get_DBAdaptor($species_name, 'otherfeatures') if !defined($ofhost);
-      if (defined $otherf_dba) { $otherf_dba->dnadb($core_dba); }
+  $species_name //= shift @{$id2name{$species_id}};
+
+  # prepare registry and core/otherfeatures dba
+  my $registry = "Bio::EnsEMBL::Registry";
+  my ($core_dba, $otherf_dba);
+
+  # for ensembl project, use provided connection details or default to staging
+  if ( $file_params->{project} eq ('ensembl') ) {
```

Comment:
What is the reason for brackets around 'ensembl' here? I'd need someone with considerably better Perl knowledge than mine to clarify but to me, it looks like it makes the second argument to _eq_ a list.

Reply:
it's harmless but unnecessary, so removed it